### PR TITLE
Handle OpenID discovery URL errors a little nicer when creating/editing sources

### LINKED ISSUE #23283: 500 Server Error after adding openid connect provider
### Description

Instead of an informative error message i get an 500 Server Error.

Log:
.../providers_openid.go:42:CreateGothProvider() [W] [640291c5] Failed to create OpenID Connect Provider with name 'xxx' with url 'openid.xxx': Get "openid.xxx": unsupported protocol scheme ""

I think it would be possible to give this info in the Webinterface

The next 500 Server Error happens if the discovery URL returns HTML content.

### Gitea Version

1.18.5

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

i use an release from Github on Uberspace
see this Guide for more info:
https://lab.uberspace.de/guide_gitea/

### Database

MySQL

### PR DESCRIPTION
**Pull request by zeripath**
When there is an error creating a new openIDConnect authentication source try to handle the error a little better.

Close #23283

### CODE REVIEW

**Comment by zeripath** [comment]
could be backported to 1.19

**Review by silverwind — APPROVED** [review]

File: templates/admin/auth/source/oauth.tmpl [hunk 1]
```diff
@@ -24,7 +24,7 @@
 		<label for="oauth2_icon_url">{{.locale.Tr "admin.auths.oauth2_icon_url"}}</label>
 		<input id="oauth2_icon_url" name="oauth2_icon_url" value="{{.oauth2_icon_url}}">
 	</div>
-	<div class="open_id_connect_auto_discovery_url required field">
+	<div class="open_id_connect_auto_discovery_url required field{{if .Err_DiscoveryURL}} error{{end}}">
```
Comment (delvh) [inline templates/admin/auth/source/oauth.tmpl:27]:
```suggestion
	<div class="open_id_connect_auto_discovery_url required field {{if .Err_DiscoveryURL}}error{{end}}">
```
As I've understood, we've now agreed not save this one space anymore to make it more readable.

**Review by delvh — APPROVED** [review]

File: templates/admin/auth/source/oauth.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/admin/auth/source/oauth.tmpl:27]: I don't think we specifically agreed to one form, but accept both forms. I still prefer the "correct" form :)

Comment (delvh) [reply templates/admin/auth/source/oauth.tmpl:27]: You mean the form without the extra space?
Same.
However, I can also see that it makes reading the code much harder.

> I don't think we specifically agreed to one form, but accept both forms

https://docs.gitea.io/en-us/guidelines-frontend/#styles-and-attributes-in-go-html-template

Comment (silverwind) [reply templates/admin/auth/source/oauth.tmpl:27]: Yes, I prefer the form that does not emit whitespace. It's slightly less readable, but still acceptable to me to not have ugly HTML in the inspector.

I see it was recommended, but that doesn't mean the other form is strictly forbidden.

**Pull request merged**